DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                            )
+                            OUTPUT Inserted.ID
+                            VALUES
+                            (
+                                @primaryEmail,
+                                @passwordHash,
+                                @firstName,
+                                @lastName,
+                                @jobGroupID,
+                                @professionalRegistrationNumber,
+                                @profileImage,
+                                @active,
+                                @resetPasswordID,
+                                NULL,
+                                0,
+                                @hasBeenPromptedForPrn,
+                                @learningHubAuthId,
+                                @hasDismissedLhLoginWarning,
+                                GETUTCDATE(),
+                                GETUTCDATE()
+                            )",
+                new
+                {
+                    primaryEmail = setEmailToGuid
+                        ? Guid.NewGuid().ToString()
+                        : delegateAccount.Email,
+                    passwordHash = delegateAccount.OldPassword ?? "",
+                    firstName = delegateAccount.FirstName_deprecated ?? "",
+                    lastName = delegateAccount.LastName_deprecated,
+                    jobGroupId = delegateAccount.JobGroupId_deprecated,
+                    professionalRegistrationNumber = delegateAccount.ProfessionalRegistrationNumber_deprecated,
+                    profileImage = delegateAccount.ProfileImage_deprecated,
+                    active = delegateAccount.Active,
+                    resetPasswordId = delegateAccount.ResetPasswordId_deprecated,
+                    hasBeenPromptedForPrn = delegateAccount.HasBeenPromptedForPrn_deprecated,
+                    learningHubAuthId = delegateAccount.LearningHubAuthId_deprecated,
+                    hasDismissedLhLoginWarning = delegateAccount.HasDismissedLhLoginWarning_deprecated,
+                }
+            );
+
+            UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(connection, userId, delegateAccount.Id);
+        }
+
+        private static void UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(
+            IDbConnection connection,
+            int userId,
+            int delegateAccountId
+        )
+        {
+            connection.Execute(
+                @"UPDATE DelegateAccounts
+                            SET
+                                UserID = @userId,
+                                Email = NULL,
+                                CentreSpecificDetailsLastChecked = GETUTCDATE()
+                            WHERE ID = @delegateAccountId",
+                new
+                {
+                    userId,
+                    delegateAccountId
+                }
+            );
+        }
+
+        private class DelegateAccount

```

Comment:
It's very unlikely, but I wonder if in the future, when we have defined other classes for these tables in the Data project's models, if someone is going to accidentally put a reference to these classes in their code, rather than the other classes. Do we think we should rename these to make it obvious these are only used in this migration? e.g. `DelegateAccountForUarMigration`?

Reply:
I mean they are private to a Migration, I would hope that no-one would do that, and if for some stupid reason they did, then it would be caught by the review process.

Reply:
Good point. I completely forgot that that was thing. Nevermind.